test/fuzzyDateTime.js

```diff
@@ -0,0 +1,248 @@
+					invocations.push(text);
+					actualDeltas.push(opt.nextUpdateInMilliseconds);
+					return { stopUpdates: true };
+				};
+				const d = new Date(2015, 8, 23, 14, 5, 29);
+				const deltas = [
+					0,
+					1 * second,
+					2 * second,
+					4 * second,
+					8 * second,
+					16 * second,
+					1 * minute,
+					1 * minute,
+				];
+
+				let totalDelta = 0;
+				for (const delta of deltas) {
+					totalDelta += delta;
+					formatFuzzyDate(d, {
+						origin: new Date(origin.getTime() + totalDelta),
+						onUpdate
+					});
+					await sleep(10);
+				}
+				formatFuzzyDate(d, {
+					origin: new Date(origin.getTime() + 4 * day),
+					onUpdate,
+				});
+
+				deltas.push(1 * minute);
+				deltas.push(-1);
+				expect(actualDeltas).to.deep.equal(deltas.slice(1));
+				expect(invocations).to.deep.equal([
+					'1 second ago',
+					'2 seconds ago',
+					'4 seconds ago',
+					'8 seconds ago',
+					'16 seconds ago',
+					'1 minute ago',
+					'2 minutes ago',
+					'3 minutes ago',
+					'9/23/2015',
+				]);
+			});
+		});
+
+		it('uses absolute formatting for recent times when rtf is not available', () => {
+			expect(formatFuzzyDate(past.seconds(5), { origin: noon })).to.not.match(shortDateRegex);
+			expect(formatFuzzyDate(past.seconds(5), { origin: noon, rtf: false })).to.match(shortDateRegex);
+		});
+
+		relativeExamples.forEach(([makeTime, expected,, { rtf, origin = noon } = {}]) => {
+			it(`formats ${label(makeTime)} as ${expected}`, () => {
+				expect(formatFuzzyDateTime(makeTime(), { origin, rtf })).to.equal(expected);
+			});
+		});
+
+		it('formats > 3 days as absolute', () => {
+			expect(formatFuzzyDateTime(past.days(3.5))).to.match(shortDateTimeRegex);
+			expect(formatFuzzyDateTime(future.days(3.5))).to.match(shortDateTimeRegex);
+		});
+
+		const sleep = ms => new Promise(resolve => setTimeout(resolve, ms));
+
+		for (const [makeTime,, expected, { origin = noon } = {}] of relativeExamples) {
+			it(`updates ${label(makeTime)} in ${label(expected)}`, (done) => {
+				formatFuzzyDateTime(makeTime(), {
+					origin,
+					onUpdate: (text, opt) => {
+						expect(opt.nextUpdateInMilliseconds)
+							.to.be.closeTo(expected(), 100);
+						done();
+						return { stopUpdates: true };
+					}
+				});
+			});
+		}
+
+		it('optionally provides updates via a callback', (done) => {
+			let callCount = 0;
+			const expectedText = ['1 second ago', '2 seconds ago'];
+			const expectedTimeout = [1000, 2000];
+			const immediateResult = formatFuzzyDateTime(past.seconds(1, new Date()), {
+				onUpdate: (text, opt = {}) => {
+					expect(text).to.equal(expectedText[callCount]);
+					expect(opt.nextUpdateInMilliseconds).to.be.closeTo(expectedTimeout[callCount], 100);
+					callCount += 1;
+					if (callCount === expectedText.length) {
+						done();
+					}
+				}
+			});
+			expect(immediateResult).to.equal(expectedText[0]);
+		}).timeout(3000);
+
+		it('the next update can be cancelled', (done) => {
```

Comment:
Converting the async tests to use the `done` callback instead of `async`/`await` seems to have fixed the race condition. I ran the tests 10+ times locally and they always pass now.

Don't know why it's still failing on travis. The error is:

```
1) number
[test]        parseNumber
[test]          edge
[test]            should stop on first invalid char "1b":
[test]      Uncaught Error: Uncaught AssertionError: expected '4 seconds ago' to equal undefined (http://localhost:8080/node_modules/chai/chai.js:239)
```

I didn't change either `number.js` or it's test file. According to the test:

```js
[
	{val: '1b', expected: 1},
	{val: '1.44e', expected: 1.44},
	{val: ' - 0.2ab09', expected: -0.2},
	{val: '4,593  	329.2b392-', expected: 4593329.2}
].forEach(function(input) {
	it(`should stop on first invalid char "${input.val}"`, () => {
		const value = parseNumber(input.val);
		expect(value).to.equal(input.expected);
	});
});
```

the expected value should be `1` not `undefined`.